website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+---
+{
+    "TableOfContents": {
+        "Name": "1.2 - Hello Quad",
+        "Url": "2-hello-quad.html",
+        "Metadata": {
+            "AuthorGitHub": "ohtrobinson",
+            "DateTimeWritten": "02/11/2022 12:00",
+            "PreviewImage": ""
+        }
+    }
+}
+---
+
+# 1.2 - Hello Quad
+<?# Info "You can view the source code for this tutorial [here.](../sources/1.2-final-result.html) This tutorial builds on the previous tutorial. If you haven't read it, you can do so [here.](1-hello-window.html)" /?>
+
+Let's draw something on-screen! In this tutorial, you'll learn:
+
+* How to initialize OpenGL.
+* How to clear the window.
+* How the OpenGL pipeline works.
+* What vertex array objects, vertex buffers, and element array buffers are, and how to use them.
+* How to create and use shaders.
+* How to draw to the screen.
+
+This tutorial will feature a lot of content and new info, so we'll take it slow and with lots of explanations.
+
+## Initializing OpenGL
+The first thing we need to do before we can issue OpenGL commands, is to initialize it. Fortunately, Silk.NET windowing does most of the hard part for us, and we only have to get the GL api.
+
+Add the following using directive to the top of your file, you'll need this to use OpenGL:
+
+```cs
+using Silk.NET.OpenGL;
+```
+
+Then, at the top of your class, add the following:
+
+```cs
+private static GL _gl;
+```
+
+Finally, in your `OnLoad` method, add the following:
+
+```cs
+_gl = _window.CreateOpenGL();
+```
+
+What are we doing here? Silk.NET requires you to keep a **reference** to the OpenGL API. If you've used or seen OpenGL in C, you'll notice that this is different to the way that it is done there. This is done so that you can more easily keep track of multiple contexts. If you don't know what that is, don't worry about it for now, we won't be using it in these tutorials.
+
+<?# Info "If you take a look at the source in the Silk.NET samples repository, you will notice that it uses `_gl = GL.GetApi(_window)`. This is another way to get the GL API, however when using Silk.NET windowing, it's recommended that you use `_window.CreateOpenGL()` instead." /?>
+
+Now, run your application again. If all is good, you should see no change. Awesome! Let's do our first steps in OpenGL: Clearing the window.
+
+## Clearing the window
+Before we start this, let's take a look at what makes up an OpenGL window. 
+
+A window consists of several **framebuffers**. A framebuffer is essentially a texture that can be drawn to by OpenGL. An OpenGL window will consist of the following framebuffers:
+
+* Color buffer
+* Depth stencil buffer
+
+On top of this, a window will often contain at least two of these buffers. This is known as **double-buffering** and is imperitive for VSync to work properly. One buffer is displayed, while another is rendered to. 
```

Comment:
I don't think double-buffering is specifically about VSync. If you're always rendering to the same framebuffer, then what will the screen display? The resulting frame will only exist for a brief period of time between you finishing your draw calls for that frame and you clearing the framebuffer to start the next. You can't display a framebuffer that's currently being worked on!

Reply:
```suggestion
On top of this, a window will often contain at least two sets of these buffers. This is known as **double-buffering** and is imperitive for VSync to work properly. One buffer is displayed, while another is rendered to. 
```

Reply:
Yeah you're right, hadn't really thought about it outside of vsync. I'll change this to make it clearer tomorrow.

Reply:
I've removed the bits about vsync, and replaced it to say "imperative for rendering to work properly."